### PR TITLE
Docs

### PR DESCRIPTION
**Pull request by purplefox**

### CODE REVIEW

**Comment by purplefox** [comment]
I thought I had already submitted this some time back, but I forgot to. Please review! :)

@Narigo @vietj

File: src/main/java/io/vertx/ext/asyncsql/package-info.java [hunk 1]
```diff
@@ -13,16 +13,89 @@
...
+ * ----
+ * {@link examples.Examples#example4}
+ * ----
+ *
+ * Once you've finished with the connection make sure you close it afterwards.
+ *
+ * The connection is an instance of {@link io.vertx.ext.sql.SqlConnection} which is a common interface used by
+ * more than Vert.x sql service.
+ *
+ * You can learn how to use it in the http://foobar[common sql interface] documentation.
```
Comment (Narigo) [inline src/main/java/io/vertx/ext/asyncsql/package-info.java]: How to link this?

File: src/main/java/examples/Examples.java [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+        // Started OK - now ready to use!
+      } else {
+        // Failed to start
+      }
+    });
+
+  }
+
+  public void example4( AsyncSqlService service) {
```
Comment (Narigo) [inline src/main/java/examples/Examples.java]: Space before param type?

File: src/main/java/io/vertx/ext/asyncsql/package-info.java (hunk 1 above)
Comment (Narigo) [inline src/main/java/io/vertx/ext/asyncsql/package-info.java]: I hope there is documentation about how to use transactions then, etc. :)

File: src/main/java/examples/Examples.java (hunk 2 above)
Comment (purplefox) [inline src/main/java/examples/Examples.java]: ack

File: src/main/java/io/vertx/ext/asyncsql/package-info.java (hunk 1 above)
Comment (purplefox) [inline src/main/java/io/vertx/ext/asyncsql/package-info.java]: tbh I have no idea how to link outside - I think this is a TODO in docgen

File: src/main/java/io/vertx/ext/asyncsql/package-info.java [hunk 3]
```diff
@@ -13,16 +13,89 @@
...
+ *
```
Comment (purplefox) [inline src/main/java/io/vertx/ext/asyncsql/package-info.java]: Documentation on usage of the common sql api is in the vertx-sql-common project

**Pull request merged**